apps/condo/domains/resident/schema/RegisterServiceConsumerService.js

```diff
@@ -223,7 +228,13 @@ const RegisterServiceConsumerService = new GQLCustomSchema('RegisterServiceConsu
                 if (!accountNumber || accountNumber.length === 0) {
                     throw new GQLError(ERRORS.ACCOUNT_NUMBER_IS_NOT_SPECIFIED, context)
                 }
-                const resident = await Resident.getOne(context, { id: residentId, deletedAt: null }, 'id user { id }')
+                const isAdmin = get(context, 'authedItem.isAdmin')
```

Comment:
isAdmin is a bad check as we can change access later
you need to add logic for Resident user